pkg/controller/daemon/daemoncontroller.go

```diff
@@ -765,14 +784,14 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 	if dsNeedsSync && ds.DeletionTimestamp == nil {
 		switch ds.Spec.UpdateStrategy.Type {
 		case extensions.RollingUpdateDaemonSetStrategyType:
-			err = dsc.rollingUpdate(ds)
+			err = dsc.rollingUpdate(ds, nodeToDaemonPods)
```

Comment:
No, DaemonSet controller didn't work that way. In the rollingUpdate case, we may create/kill some pods after `manage`, and then do the rolling update based on `nodeToDaemonPods`. If we use the cached value after we changed those pods in `manage`, rolling update may create or kill more pods than needed. It might be fine for Deployments, but it's safer not to do this in DaemonSet controller (since every node expects one and exactly one pod.)